pkg/apiserver/handlers.go

```diff
@@ -188,11 +190,11 @@ func (t *timeoutHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
-	done := make(chan struct{}, 1)
+	done := make(chan struct{})
 	tw := newTimeoutWriter(w)
 	go func() {
 		t.handler.ServeHTTP(tw, r)
```

Comment:
Yeah, it makes it more obvious to monitoring tools. It also means that
someone who figures out how to trigger one can potentially DOS apiserver--
for this reason we did leave the mechanism in place and it can be turned on
by setting kruntime.ReallyCrash = false. (So you probably should defer
HandleCrash to keep that working...)

On Tue, Aug 9, 2016 at 10:55 AM, krousey <notifications@github.com> wrote:

> In pkg/apiserver/handlers.go
> <https://github.com/kubernetes/kubernetes/pull/29594#discussion_r74110174>
> :
>
> > @@ -188,11 +190,11 @@ func (t *timeoutHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
> >  		return
> >  	}
> >
> > -	done := make(chan struct{}, 1)
> > +	done := make(chan struct{})
> >  	tw := newTimeoutWriter(w)
> >  	go func() {
> >  		t.handler.ServeHTTP(tw, r)
>
> Oh so the crash is desired. Ok.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/kubernetes/kubernetes/pull/29594/files/b9129837e9db2272de6f73ba88552451ae1438de#r74110174>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAnglkqF0mNAB8rRVGWU-W1HG82nHT1oks5qeL8HgaJpZM4JUwp4>
> .
>